### ISSUE TITLE
Force linter for verilog_systemverilog filetype

### ISSUE THREAD

**Issue by SiliconKite** [w0rp/ale#1674]
## Information

VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Oct  2 2017 11:16:01)
Included patches: 1-1171
Fedora 25

### :ALEInfo

 Current Filetype: verilog_systemverilog
Available Linters: []
  Enabled Linters: []
 Linter Variables:

 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = v:null
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 0
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'never'
let g:ale_lint_on_insert_leave = 0
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let b:ale_linters = {'systemverilog': ['iverilog']}
let g:ale_linters_explicit = 0
let g:ale_list_window_size = v:null
let g:ale_list_vertical = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = {}
let g:ale_pattern_options_enabled = 0
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = v:null
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_statusline_format = v:null
let g:ale_type_map = v:null
let g:ale_use_global_executables = v:null
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
let b:ale_warn_about_trailing_whitespace = 1

## What went wrong

Not sure this a bug, feature request or misunderstanding from my side (most likely the last one). Anyway, I am trying to get ALE to work as 'systemverilog' linter on a filetype of 'verilog_systemverilog' type. This filetype is enabled when installing the following plugin https://github.com/vhda/verilog_systemverilog.vim. Both syntaxis are pretty much the same so I am assuming it is possible to do this ?
Basically  I am doing  let b:ale_linters = {'systemverilog' : ['iverilog'],} when in the "verilog_syntaxverilog' file to enable the ALE linter.

## Reproducing the bug

1. install verilog_systemverilog syntax file (it can be done via the plugin or not)

2. Open a .sv file, do b:ale_linters = {'systemverilog' : ['iverilog'],} when in the "verilog_syntaxverilog' file.

Also worth mentioning is that if I open a .sv file, and do a syntax=systemverilog, ALE plugin is not working either ...

Any help much appreciated
Thanks

**Comment by w0rp** [comment]
`'iverilog'` isn't a filetype that ALE has linters for, but `'verilog'` is. Try updating your configuration. If it works well, I might update the defaults so files with the filetype `systemverilog` load `verilog` linters by default.

**Comment by w0rp** [comment]
Okay, I'll set this alias up by default, so others won't have the same issue.
